doc/source/dev/devsetup.rst

```diff
@@ -0,0 +1,35 @@
+Development Setup
+=================
+
+.. note::
+
+   To make changes to a backend, e.g. to the way a simulator or synthesis tool is called, you need to edalize, not fusesoc.
+   Edalize is a separate project, see https://github.com/olofk/edalize for more information.
+
+Linux
+-----
+
+.. note::
+
+   If you have already installed FuseSoC, remove it first using ``pip3 uninstall fusesoc``.
+
```

Comment:
Why? Because pip won't update by default? We could suggest install -U instead. Same end result though. Or is it when you want to install with -e for development. Don't think you need to uninstall in that case. Your way works fine. Just want to check I didn't miss anything

Reply:
This ensures that only a single installation remains after executing these steps.

The problem are per-user vs global installations. pip uninstall will remove all existing installations, both global and per-user ones. pip install -U is less reliable. Think of the following scenario:

```
pip3 install -U --user fusesoc
sudo pip3 install -U fusesoc

# now you have two installations, one in /usr/local, and one in ~/.local
```

I've seen a fair amount of people confused by that behavior, and pip uninstall is not much harder but slightly more reliable.